Automattic/jetpack
# Docker: Make SFTP key auth easy, and adding plugins/themes less likely to break Backup/Scan

#### Changes proposed in this Pull Request:

* Adds a `docker/data/ssh.keys/`, into which ssh public key files can be placed to allow key-based login for SFTP.
* Adds a `docker/compose-volumes.yml` file for defining the bind mounts for WordPress and its plugins and themes, so these won't get out of sync between the web service container and the sftp container.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

I'll assume you're already reasonably familiar with the docker-based dev environment.

* `yarn docker:down`
* Generate an ssh keypair, and copy the public key into `docker/data/ssh.keys/`. Filename doesn't matter as long as it ends with `.pub`.
* `yarn docker:compose-volumes` if necessary to generate a default `docker/compose-volumes.yml`. Or copy `docker/compose-volumes.yml.sample` manually.
* Add a plugin or theme by editing `docker/compose-volumes.yml` following the instructions in comments in that file.
* `yarn docker:up`
* Verify your plugin or theme is available in WordPress's UI.
* Connect to the container using sftp and the private key generated earlier.
    * On a command line, that'd probably be something like `sftp -o IdentitiesOnly=yes -i /path/to/key -P 1022 wordpress@localhost`.
* When connected, see that your added plugin or theme can be found via sftp.

#### Proposed changelog entry for your changes:
* I see previous changes to the Docker env haven't done changelog entries, so I guess this won't either

#### Note
* This is my first attempt at a PR, please let me know if I did anything wrong!
